### PR TITLE
Adds support for 'signed' secureboot files that may be used during manufacturing

### PR DESCRIPTION
**Pull request by Flickdm**
This PR adds support for "signed" secure boot objects that works with API calls like `SetFirmwareVariableEx(..)` and powershell commands like `Set-SecureBootUefi`

This matches the behavior by [CreateTimeBasePayload(..)](https://github.com/microsoft/mu_tiano_plus/blob/7acba38909add22c71b3c133ba01fec85096c14a/SecurityPkg/Library/SecureBootVariableLib/SecureBootVariableLib.c#L267) which creates EFI_VARIABLE_AUTHENTICATION_2 descriptor with an empty signature

The empty signature works because of this [code](
https://github.com/microsoft/mu_tiano_plus/blob/7acba38909add22c71b3c133ba01fec85096c14a/SecurityPkg/Library/AuthVariableLib/AuthService.c#L728)

In order to not cause a breaking change, the decision to leave firmware architecture specific files were left at the top level and an additional "Imaging" folder created

```txt
Artifacts:
|   README.md
|
+---Aarch64
|       Default3PDb.bin
|       DefaultDb.bin
|       DefaultDbx.bin
|       DefaultKek.bin
|       DefaultPk.bin
|
+---Arm
|       Default3PDb.bin
|       DefaultDb.bin
|       DefaultDbx.bin
|       DefaultKek.bin
|       DefaultPk.bin
|
+---Ia32
|       Default3PDb.bin
|       DefaultDb.bin
|       DefaultDbx.bin
|       DefaultKek.bin
|       DefaultPk.bin
|
+---Imaging
|   +---Aarch64
|   |       Default3PDb.bin
|   |       DefaultDb.bin
|   |       DefaultDbx.bin
|   |       DefaultKek.bin
|   |       DefaultPk.bin
|   |
|   +---Arm
|   |       Default3PDb.bin
|   |       DefaultDb.bin
|   |       DefaultDbx.bin
|   |       DefaultKek.bin
|   |       DefaultPk.bin
|   |
|   +---Ia32
|   |       Default3PDb.bin
|   |       DefaultDb.bin
|   |       DefaultDbx.bin
|   |       DefaultKek.bin
|   |       DefaultPk.bin
|   |
|   \---X64
|           Default3PDb.bin
|           DefaultDb.bin
|           DefaultDbx.bin
|           DefaultKek.bin
|           DefaultPk.bin
|
\---X64
        Default3PDb.bin
        DefaultDb.bin
        DefaultDbx.bin
        DefaultKek.bin
        DefaultPk.bin
```

### CODE REVIEW

File: scripts/secure_boot_default_keys.py [hunk 1]
```diff
@@ -161,6 +166,72 @@ def _convert_csv_to_signature_list(file: str, signature_owner: str, target_arch:
...
 
 
+def _convert_to_empty_auth_var(esl_payload: bytes) -> bytes:
+    """This function is used to add a empty authenticated variable header to a secure boot efi signature list.
+
+    Args:
+        esl_payload: The secure boot efi signature list
+
+    Returns:
+        bytes: The wrapped empty authenticated secure boot efi signature list
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
        (bytes): The wrapped empty authenticated secure boot efi signature list

```

File: scripts/secure_boot_default_keys.py [hunk 2]
```diff
@@ -161,6 +166,72 @@ def _convert_csv_to_signature_list(file: str, signature_owner: str, target_arch:
     return siglist.encode()
 
 
+def _convert_to_empty_auth_var(esl_payload: bytes) -> bytes:
+    """This function is used to add a empty authenticated variable header to a secure boot efi signature list.
+
+    Args:
+        esl_payload: The secure boot efi signature list
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]: nit: The rest of the file is not correct, but we don't generate documentation so it's not a big enough of a deal to correct the untouched code.

```suggestion
        esl_payload (bytes): The secure boot efi signature list

```

File: scripts/secure_boot_default_keys.py [hunk 3]
```diff
@@ -331,12 +438,31 @@ def create_readme(keystore: dict, arch: str) -> str:
     return bytes(readme, "utf-8")
 
 
+def create_folder(directory: Path) -> None:
+    """Creates a folder if it does not exist.
+
+    Args:
+        directory: The path to the folder to create.
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
        directory (Path): The path to the folder to create.

```

File: scripts/secure_boot_default_keys.py [hunk 4]
```diff
@@ -331,12 +438,31 @@ def create_readme(keystore: dict, arch: str) -> str:
     return bytes(readme, "utf-8")
 
 
+def create_folder(directory: Path) -> None:
+    """Creates a folder if it does not exist.
+
+    Args:
+        directory: The path to the folder to create.
+
+    Returns:
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]: `Returns` part can be deleted for functions that return None

File: scripts/secure_boot_default_keys.py [hunk 5]
```diff
@@ -364,18 +490,30 @@ def main() -> int:
         for key, value in default_keys.items():
             arch, variable = key
 
-            out_dir = Path(args.output, arch.capitalize())
+            parent = Path(args.output)
 
-            out_dir.mkdir(exist_ok=True, parents=True)
-            out_dir.touch()
+            firmware_architecture = Path(parent, arch.capitalize())
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
            firmware_architecture = parent / arch.capitalize()

```

File: scripts/secure_boot_default_keys.py [hunk 6]
```diff
@@ -364,18 +490,30 @@ def main() -> int:
...
-            out_dir = Path(args.output, arch.capitalize())
+            parent = Path(args.output)
 
-            out_dir.mkdir(exist_ok=True, parents=True)
-            out_dir.touch()
+            firmware_architecture = Path(parent, arch.capitalize())
+            create_folder(firmware_architecture)
 
-            out_file = Path(out_dir, f"{variable}.bin")
+            out_file = Path(firmware_architecture, f"{variable}.bin")
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
            out_file = firmware_architecture / f"{variable}.bin"

```

File: scripts/secure_boot_default_keys.py [hunk 7]
```diff
@@ -364,18 +490,30 @@ def main() -> int:
...
             if out_file.exists():
                 out_file.unlink()
             with open(out_file, "wb") as f:
                 f.write(value)
 
-            readme_path = Path(out_dir, "README.md")
+            imaging_dir = Path(parent, "Imaging")
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
            imaging_dir = parent / "Imaging"

```

File: scripts/secure_boot_default_keys.py [hunk 8]
```diff
@@ -364,18 +490,30 @@ def main() -> int:
...
+            create_folder(imaging_dir)
+
+            imaging_architecture = Path(imaging_dir, arch.capitalize())
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
            imaging_architecture = imaging_dir / arch.capitalize()

```

File: scripts/secure_boot_default_keys.py [hunk 9]
```diff
@@ -364,18 +490,30 @@ def main() -> int:
...
+            create_folder(imaging_architecture)
+
+            out_file = Path(imaging_architecture, f"{variable}.bin")
```
Comment (Javagedes) [inline scripts/secure_boot_default_keys.py]:
```suggestion
            out_file = imaging_architecture / f"{variable}.bin"

```

**Review by Javagedes — APPROVED** [review]
looks good. I had a small nit with the function documentation, and then just an alternative in path forming. Not that we care about performance, but you technically don't need to use the `Path` construcntor, the `/` function has been overwritten for the `Path` object to do something similar. You can mark as won't fix if you don't want to make the change, it's nbd.

**Review by apop5 — APPROVED** [review]

**Pull request merged**